app/src/androidTest/java/com/github/orkest/bluetooth/data/OrkestBluetoothCommunicationTest.kt

```diff
@@ -18,57 +21,75 @@ class OrkestBluetoothCommunicationTest {
     private lateinit var testSocket: TestSocket
 
     private lateinit var msgReceived: String
+    private lateinit var msgSent: String
 
-    private val handler = object : Handler(Looper.getMainLooper()) {
-        override fun handleMessage(msg: Message) {
-            when (msg.what) {
 
-                BluetoothConstants.MESSAGE_WRITE -> {
-                    // construct a string from the buffer
-                    println("Message write")
-                    msgReceived = String(msg.obj as ByteArray)
-                }
-                BluetoothConstants.MESSAGE_READ -> {
-                    // construct a string from the valid bytes in the buffer
-                    msgReceived = String(msg.obj as ByteArray, 0, msg.arg1)
-                }
-                BluetoothConstants.MESSAGE_TOAST -> {
-                    msgReceived= msg.data.getString("toast").toString()
+    private lateinit var handler : Handler
+    @Before
+    fun setup() {
+        msgReceived = ""
+        msgSent = ""
+        handler = object : Handler(Looper.getMainLooper()) {
+            override fun handleMessage(msg: Message) {
+                when (msg.what) {
+
+                    BluetoothConstants.MESSAGE_WRITE -> {
+                        // construct a string from the buffer
+                        println("Message write")
+                        msgSent = String(msg.obj as ByteArray)
+                    }
+                    BluetoothConstants.MESSAGE_READ -> {
+                        // construct a string from the valid bytes in the buffer
+                        msgReceived = String(msg.obj as ByteArray, 0, msg.arg1)
+                    }
+                    BluetoothConstants.MESSAGE_TOAST -> {
+                        msgReceived= msg.data.getString("toast").toString()
+                    }
```

Comment:
Use log.d() instead of print.

Your initialization = "", could be refactored in a separate function as it is used multiple times.